### PR TITLE
removes unnecessary checks

### PR DESCRIPTION
**Pull request by isandeep41**
Fixes unnecessary checks

<hr>

This PR has:
- [x] been self-reviewed.
   - [x] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [x] added documentation for new or modified features or behaviors.
- [x] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [x] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [x] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [x] added integration tests.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

File: extensions-core/lookups-cached-single/src/main/java/org/apache/druid/server/lookup/PollingLookup.java [hunk 1]
```diff
@@ -132,9 +132,7 @@ public String apply(@Nullable String key)
       return NullHandling.emptyToNullIfNeeded((String) cache.get(keyEquivalent));
     }
     finally {
-      if (cacheRefKeeper != null && cache != null) {
```
Comment (asdf2014) [inline extensions-core/lookups-cached-single/src/main/java/org/apache/druid/server/lookup/PollingLookup.java:135]: I believe `cache != null` is necessary

File: extensions-core/lookups-cached-single/src/main/java/org/apache/druid/server/lookup/PollingLookup.java [hunk 2]
```diff
@@ -162,9 +160,7 @@ public String apply(@Nullable String key)
       return cache.getKeys(valueEquivalent);
     }
     finally {
-      if (cacheRefKeeper != null && cache != null) {
```
Comment (asdf2014) [inline extensions-core/lookups-cached-single/src/main/java/org/apache/druid/server/lookup/PollingLookup.java:165]: Same

**Review by asdf2014 — CHANGES_REQUESTED** [review]

File: extensions-core/lookups-cached-single/src/main/java/org/apache/druid/server/lookup/PollingLookup.java (hunk 1 above)
Comment (isandeep41) [reply extensions-core/lookups-cached-single/src/main/java/org/apache/druid/server/lookup/PollingLookup.java:135]: it seems the code block is not dependent on cache, but I have included the `cache != null` check

File: extensions-core/lookups-cached-single/src/main/java/org/apache/druid/server/lookup/PollingLookup.java (hunk 2 above)
Comment (isandeep41) [reply extensions-core/lookups-cached-single/src/main/java/org/apache/druid/server/lookup/PollingLookup.java:165]: fixed

**Review by asdf2014 — APPROVED** [review]
LGTM, thanks for your contribution

**Review by suneet-s — APPROVED** [review]
👍

**Comment by suneet-s** [comment]
Added static analysis tag since intelliJ says these conditions are always true

**Pull request merged**